my-code/whats-in/src/App.jsx

```diff
@@ -0,0 +1,90 @@
+import { Routes, Route, useNavigate } from 'react-router-dom';
+import React, { useEffect, useState } from 'react'
+import Homepage from './pages/Homepage'
+import Moviepage from './pages/Movie'
+import Header from './components/Header'
+
+function App() {
+  const [searchText, setSearchText] = useState('')
+  const [movies, setMovies] = useState()
+  const [selectedMovie, setSelectedMovie] = useState(null)
+  const [selectedMovieInfo, setSelectedMovieInfo] = useState()
+  const [isLoading, setIsLoading] = useState(false)
+  const [likedMovies, setLikedMovies] = useState([]);
+  const [isEmptySearch, setIsEmptySearch] = useState(true)
+  const navigate = useNavigate()
+  
+  const handleSearch = (text) => {
+    setSearchText(text)
+  }
+
+  useEffect(() => {
+    fetch(`https://www.omdbapi.com/?s=${searchText}&apikey=${process.env.REACT_APP_API_KEY}`)
+    .then(response => response.json())
+    .then(data => {
+      if (searchText === '') {
+        setIsEmptySearch(true)
+        setMovies('')
+      } else {
+        setIsEmptySearch(false)
+        setMovies(data.Search)
+      }
+    })
+    .catch(error => {
+      console.error(error);
+    })
+  }, [searchText])
+
+  const handleMovieSelection = (movie) => {
+    setSelectedMovie(movie)
+  }
+
+  useEffect(() => {
+    if (selectedMovie && selectedMovie.imdbID) {
+      setIsLoading(true)
+      fetch(`https://www.omdbapi.com/?i=${selectedMovie.imdbID}&apikey=${process.env.REACT_APP_API_KEY}`)
+        .then(response => response.json())
+        .then(data => {
+          setSelectedMovieInfo(data)
+          navigate('/movie')
+          setSelectedMovie(null)
+        })
```

Comment:
again, an optimization could be made by simply inferring from the data we have.
it's unnecessary (and objectively worse because you have to be more careful and you're making it possible to have inconsistent data: what happens if you have `selectedMovie` but no `selectedMovieInfo`?) to have both this states. You could infer that you have a movie selected by checking if you have data in `selectedMovieInfo`.

Also, as stated above, you're navigating as a side-effect of loading movie data, where it should be:
A) no navigation and render components based on available data
B) exactly the opposite and simply navigate and fetch because of the route change